### PR TITLE
#5688 break up into package for each kernel, python package 

### PR DESCRIPTION
**Pull request by rbidas**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,13 +15,28 @@ jupyter-groovy-kernel.log
...
 /gradle*
-beaker.egg-info/
+*.egg-info/
 
 beakerx/static/
-beakerx.egg-info/
 beakerx/beakerx/static/
-beakerx/beakerx/kernel/
-beakerx/dist
+beakerx-base/beakerx_base
```
Comment (scottdraves) [inline .gitignore]: put them in a directory "package" like "package/base", "package/groovy", etc.

File: README.md [hunk 2]
```diff
@@ -47,12 +47,26 @@ how to install and run BeakerX.
...
+```
+
+#### Build and install only Beakerx
+```
+python -m build --base
+```
+
+#### Build and install kernel
+```
+python -m build --kernel=<kernel_name>
```
Comment (scottdraves) [inline README.md:75]: this looks good.

File: beakerx-base/LICENSE [hunk 3]
```diff
@@ -0,0 +1,202 @@
+Apache License
```
Comment (scottdraves) [inline beakerx-base/LICENSE]: we don't need any more copies of this.  the one at the top of the repo is enough.

File: beakerx-base/NOTICE [hunk 4]
```diff
@@ -0,0 +1,5 @@
+BeakerX: Beaker Extensions for Jupyter
```
Comment (scottdraves) [inline beakerx-base/NOTICE]: ditto, not needed here

**Comment by rbidas** [comment]
Done

File: .gitignore [hunk 5]
```diff
@@ -15,13 +15,28 @@ jupyter-groovy-kernel.log
...
 /gradle*
-beaker.egg-info/
+*.egg-info/
 
 beakerx/static/
-beakerx.egg-info/
 beakerx/beakerx/static/
-beakerx/beakerx/kernel/
-beakerx/dist
+packages/base/beakerx_base
```
Comment (scottdraves) [inline .gitignore]: "package" not "packages"

File: doc/contents/EasyFormPythonDemos.ipynb [hunk 6]
```diff
@@ -13,7 +13,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "from beakerx import *"
+    "from beakerx_python import *"
```
Comment (scottdraves) [inline doc/contents/EasyFormPythonDemos.ipynb]: i don't like this but i am not sure what to do about it.

**Comment by rbidas** [comment]
Done

**Comment by rbidas** [comment]
Merge from master

File: beakerx/beakerx/install.py [hunk 7]
```diff
@@ -80,36 +74,44 @@ def _install_css():
...
+            '--sys-prefix', '--replace',
+            '--name', kernel_name, tmpdir
         ]
-        subprocess.check_call(uninstall_cmd)
+        subprocess.check_call(install_cmd)
+
+
+def uninstall_kernel(kernel_name):
+    uninstall_cmd = [
+        'jupyter', 'kernelspec', 'remove', kernel_name, '-y', '-f'
```
Comment (jmsdnns) [inline beakerx/beakerx/install.py:103]: What would happen here if I already have a `jupyter` in my path? Is care taken somewhere else to make sure the right one is called?

Comment (rbidas) [reply beakerx/beakerx/install.py:103]: No, It use jupyter from PATH. It is made the same as install kernel and nbextension

**Pull request closed without merging**